internal/validate/string.go

```diff
@@ -84,11 +84,11 @@ func (s StringRule) Validate() *Failure {
 		problems = append(problems, fmt.Sprintf(format, args...))
 	}
 	if s.MinLength > 0 && len(value) < s.MinLength {
-		add("length of string is %d, must be at least %d", len(value), s.MinLength)
+		add("%s must be at least %d characters", s.Name, s.MinLength)
```

Comment:
Repeating the field name in the message I think will end up duplicating the field name in the consolidated error message here: https://github.com/infrahq/infra/blob/main/internal/validate/validate.go#L121

We could do make a similar change for all the validators and remove it from `Error.Error`, but I kind of like the way it works now. It gives the caller the option to prepend the field name or group them differently if they choose.

What do you think about removing the field name from here, and instead prepending it in the UI ?

Otherwise the rewording of the message sounds good to me.

Reply:
My reason for rewording this was less about the field name and more about the error message being user-friendly to people without programming experience. This message is actually displayed in the UI.

`length of string is X, must be at least Y` is confusing to the average person, I'll move the name of the field up though 👍